### PR TITLE
Refactor SVGImage for dragMove warning

### PR DESCRIPTION
**Pull request by mcbouslog**
## Package
- lib-classifier

## Describe your changes
- removes development console warning `Warning: React does not recognize the "dragMove" prop on a DOM element.`

## How to Review
_Helpful explanations that will make your reviewer happy:_
- What Zooniverse project should my reviewer use to review UX?
  - https://local.zooniverse.org:8080/?project=335 (I Fancy Cats)
- What user actions should my reviewer step through to review this PR?
  - inspect console, note absence of noted warning
- Which storybook stories should be reviewed? N/A
- Are there plans for follow up PR’s to further fix this bug or develop this feature? No

# Checklist
_PR Creator - Please cater the checklist to fit the review needed for your code changes._
_PR Reviewer - Use the checklist during your review. Each point should be checkmarked or discussed before PR approval._

## General
- [x] Tests are passing locally and on Github
- [ ] Documentation is up to date and changelog has been updated if appropriate
- [x] You can `yarn panic && yarn bootstrap` or `docker-compose up --build` and FEM works as expected
- [x] FEM works in all major desktop browsers: Firefox, Chrome, Edge, Safari (Use Browserstack account as needed)
- [ ] FEM works in a mobile browser

## Refactoring
- [x] The PR creator has described the reason for refactoring
- [x] The refactored component(s) continue to work as expected

### CODE REVIEW

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGImage/SVGImage.js [hunk 1]
```diff
@@ -33,19 +33,30 @@ export default function SVGImage ({
...
+        aria-label={`Subject ${subjectID}`}
+        dragMove={move ? onDrag : () => null}
+        height={naturalHeight}
+        role='img'
+        filter={invert ? 'url("#svg-invert-filter")' : undefined}
+        width={naturalWidth}
+        href={src}
+      />
+    )
+  } else {
```
Comment (mcbouslog) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGImage/SVGImage.js:48]: Do we have a team preference between using `else` or just returning something? I'm happy to change it however preferred.

Comment (goplayoutside3) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGImage/SVGImage.js:48]: For clarity, what do you mean by just returning something? I think in this situation I would do the following. Your if else solution works, too.
```
return (
  {move ? (
    <DraggableImage />
  ) : (
    <image etc />
  )
)
```

Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGImage/SVGImage.js:48]: If you change the `dragMove` attribute in the original code, you don’t need an if block at all. `dragMove={ move ? onDrag : undefined}`. 

That way, you aren’t writing all the other component props twice. Of course, you could also define the component props once, outside the if statement.

Comment (eatyourgreens) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGImage/SVGImage.js:48]: @goplayoutside3 I think Mark’s asking whether we prefer a default return for components. 

```js
if (condition === true) {
  return conditionalValue
}
return defaultValue
```

I don’t think I’ve got a strong opinion about this. However, statements have side effects in JavaScript. Expressions don’t have side effects so I’d lean towards using a ternary expression for `dragMove`.

Comment (mcbouslog) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SVGComponents/SVGImage/SVGImage.js]: @goplayoutside3 - @eatyourgreens is correct, I was referring to if we have a team preference for a default return or using `else`.

@eatyourgreens:
- thank you for noting statements have side effects while expressions don't, not sure if I've forgotten or ever knew, but that's interesting and good to know!
- if I change the original code to `dragMove={ move ? onDrag : undefined}` I still get the noted console warning (only in dev environment)

I've made an update, combining the great suggestions, thank you both!

**Review by goplayoutside3 — APPROVED** [review]
Looks good!

I noticed while reviewing there's a second console warning related to dragging on master and this branch. For instance, when I load i-fancy-cats or the ImageToolbar storybook, then click on the MoveButton, the console displays:
```
Warning: Received `false` for a non-boolean attribute `dragging`.
```

The console also shows suggestions on how to fix it, so non-blocking suggestion to either include a fix for that warning in this dragging related PR, or follow-up in a separate PR.

**Comment by mcbouslog** [comment]
> The console also shows suggestions on how to fix it, so non-blocking suggestion to either include a fix for that warning in this dragging related PR, or follow-up in a separate PR.

I think it's a good idea to include a fix for that in this PR as well 👍  , I'll update this PR accordingly (hopefully today)...

**Comment by eatyourgreens** [comment]
The `dragging` prop is used in various components, to indicate when dragging is in progress. If you do change that here, make sure that your changes don't break existing behaviour.

Personally, I'd either ignore that warning or leave it to a second PR.

**Comment by mcbouslog** [comment]
I'll include rewriting [dragging.spec](https://github.com/zooniverse/front-end-monorepo/blob/fix-draggable-warning/packages/lib-classifier/src/plugins/drawingTools/components/draggable/draggable.spec.js) to remove enzyme, so will indeed open a subsequent PR.

**Pull request merged**